src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,122 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    int size = 0;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Object[] elementArray;
+
+    MyHashMap() {
+        elementArray = new Object[16];
+    }
+
+    MyHashMap(int capacity) {
+        elementArray = new Object[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
+        if (++size > elementArray.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = hash(key);
+
+        if (index > -1) {
+            Node currentNode = (Node) elementArray[index];
+            if (currentNode == null) {
+                elementArray[index] = new Node(key, value);
+            } else {
+                while (currentNode.next != null) {
+                    if (key.equals(currentNode.key)) {
+                        currentNode.data = value;
+                        size--;
+                        return;
+                    }
+                    currentNode = currentNode.next;
+                }
+                if (key.equals(currentNode.key)) {
+                    currentNode.data = value;
+                    size--;
+                    return;
+                }
+                currentNode.next = new Node(key, value);
+            }
 
+        } else {
+            Node currentNode = (Node) elementArray[0];
+            Node tmp = new Node(null, value);
+            if (currentNode != null) {
+                if (currentNode.key == null) {
+                    currentNode.data = value;
+                    size--;
+                    return;
+                }
+                tmp.next = currentNode;
+                elementArray[0] = tmp;
+            } else {
+                elementArray[0] = tmp;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = hash(key);
+        Node currentNode = (Node) elementArray[0];
```

Comment:
```suggestion
        Node<K, V> currentNode = elementArray[0];
```
потрібно скрізь використовувати параметризовані значення ноди

Reply:
```suggestion
        Node<K, V> currentNode = elementArray[0];
```